Identification of transcription units (TUs) encoded in a bacterial genome is essential to the elucidation of how a transcriptome is organized. Compared to the statically defined operons, TUs are defined dynamically as co-transcribed genes under different conditions. Here we use strand-specific RNA-seq (ssRNA-seq) data to derive the organization of the TUs in Clostridium thermocellum, using a machine-learning approach. Based on two parameters measuring the RNA-seq expression patterns across the genome: expression-level continuity and variance, our algorithm can accurately identify the boundaries of individual TUs. For each predicted TU, we have assessed its quality using: (i) enrichment of cis-regulatory motifs in the immediate upstream region of the TU; (ii) occurrence of transcription terminators in the immediate downstream region of the TU; and (iii) expression levels of the intergenic regions. A total of 2,619 distinct TUs are identified based on four ssRNA-seq datasets with strong assessment results. In all identified TUs, 42% have multiple genes, and the TUs on the leading strand have more genes than those on the lagging strand. In comparison of the identified TUs with predicted operons, the consistency decreases when a consecutive gene pair has the length of intergenic regions between 50 and 200 bps while the rest of gene pairs have higher consistency.